[Issue]: failed to build pytoch docker image

### Problem Description

Hi TheRock team, 
I run the below command to build theRock pytorch docker image, 
docker build -f dockerfiles/pytorch-dev/pytorch_dev_ubuntu_24.04.Dockerfile -t therock_torch_2_6_ubuntu24_04_gfx1100:latest --build-arg AMDGPU_TARGETS=gfx1100 .

but met the below issue. I checked the cmake version, which is 3.28.3, should meet the version requirement of CMake utils common module.   have you met this issue? I tried to build the standalone application of rocminfo outsides of docker building, it can work.  
246.6 [rocminfo configure] -- TheRock external project configuration injection (rocminfo)
246.6 [rocminfo configure] -- The C compiler identification is GNU 13.3.0
246.6 [rocminfo configure] -- The CXX compiler identification is GNU 13.3.0
246.6 [rocminfo configure] -- Detecting C compiler ABI info
246.6 [rocminfo configure] -- Detecting C compiler ABI info - done
246.6 [rocminfo configure] -- Check for working C compiler: /usr/bin/cc - skipped
246.6 [rocminfo configure] -- Detecting C compile features
246.6 [rocminfo configure] -- Detecting C compile features - done
246.6 [rocminfo configure] -- Detecting CXX compiler ABI info
246.6 [rocminfo configure] -- Detecting CXX compiler ABI info - done
246.6 [rocminfo configure] -- Check for working CXX compiler: /usr/bin/c++ - skipped
246.6 [rocminfo configure] -- Detecting CXX compile features
246.6 [rocminfo configure] -- Detecting CXX compile features - done
246.6 [rocminfo configure] 3.28.3
246.6 [rocminfo configure] CMake Error at CMakeLists.txt:57 (include):
246.6 [rocminfo configure]   include could not find requested file:
246.6 [rocminfo configure]
246.6 [rocminfo configure]     utils
246.6 [rocminfo configure]
246.6 [rocminfo configure]


### Operating System

ubuntu 24.04

### CPU

Intel  Xeon-D

### GPU

AMD W7900

### ROCm Version

Rocm 6.4 

### ROCm Component

_No response_

### Steps to Reproduce

pull the latest codes of TheRock, and run the below command to build docker image:
docker build -f dockerfiles/pytorch-dev/pytorch_dev_ubuntu_24.04.Dockerfile -t therock_torch_2_6_ubuntu24_04_gfx1100:latest --build-arg AMDGPU_TARGETS=gfx1100 .

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

_No response_

### Additional Information

_No response_

I tried to reproduce building commit 86cdc5a in an interactive docker. Building `rocminfo` with
```bash
AMDGPU_TARGETS=gfx1100
time cmake -GNinja -S/therock/src -B /therock/src/build-docker   -DTHEROCK_BUNDLE_SYSDEPS=ON   -DTHEROCK_ENABLE_ALL=ON   -DTHEROCK_AMDGPU_DIST_BUNDLE_NAME=custom   "-DTHEROCK_AMDGPU_TARGETS=$AMDGPU_TARGETS
cmake --build /therock/src/build-docker/ --target rocminfo
```
passes for me with the deps installed outlined in the Dockerfile. However, I see `rccl` failing with
> /therock/src/build-docker/base/rocm_smi_lib/dist/include/rocm_smi/kfd_ioctl.h:26:10: fatal error: 'libdrm/drm.h' file not found

As @stellaraccident mentioned, this was probably due to the recent changes I made and I am looking into that specific issue.

> > I tried to reproduce building commit [86cdc5a](https://github.com/ROCm/TheRock/commit/86cdc5a0a5a13cf419d231375b7bcfeb1346ce08) in an interactive docker. Building `rocminfo` with
> > AMDGPU_TARGETS=gfx1100
> > time cmake -GNinja -S/therock/src -B /therock/src/build-docker   -DTHEROCK_BUNDLE_SYSDEPS=ON   -DTHEROCK_ENABLE_ALL=ON   -DTHEROCK_AMDGPU_DIST_BUNDLE_NAME=custom   "-DTHEROCK_AMDGPU_TARGETS=$AMDGPU_TARGETS
> > cmake --build /therock/src/build-docker/ --target rocminfo
> > passes for me with the deps installed outlined in the Dockerfile. However, I see `rccl` failing with
> > > /therock/src/build-docker/base/rocm_smi_lib/dist/include/rocm_smi/kfd_ioctl.h:26:10: fatal error: 'libdrm/drm.h' file not found
> > 
> > 
> > As [@stellaraccident](https://github.com/stellaraccident) mentioned, this was probably due to the recent changes I made and I am looking into that specific issue.
> 
> Yes, that is the issue that prompted me to add those includes. I think we should add them back.
> 
> It is actually a really tricky, image by image situation to ensure you don't accidentally link against the system headers. I think I want to harden that further in the toolchain setup by always adding the sysdeps include path first (vs the more targeted where e just added it to rocm_smi). But we can start by just adding back what I had.

I'll bring it back and fix some other issues with the container. Working on it right in this moment.

HI @marbre @stellaraccident  
I just build the docker image on commit [86cdc5a] (https://github.com/ROCm/TheRock/commit/86cdc5a0a5a13cf419d231375b7bcfeb1346ce08), , but still meet the issue. I am building the docker image from scatch, and working on a bare metal machine , not inside a docker. I attached the detailed rocminfo building logs as shown below, for your debugging. Utils is a Cmake common component, which is usually related with Cmake version. but I have checked the cmake version of image building, which is 3.28.3 and should support Utils. 
Thanks

255.7 [198/303] Stage installing sub-project ROCR-Runtime
255.7 [199/303] Merging sub-project dist directory for ROCR-Runtime
256.2 [200/303] Configure sub-project rocminfo (in background)
256.2 FAILED: core/rocminfo/stamp/configure.stamp core/rocminfo/build/CMakeCache.txt core/rocminfo/build/cmake_install.cmake compile_commands_fragment_rocminfo.json /therock/build/core/rocminfo/stamp/configure.stamp /therock/build/core/rocminfo/build/CMakeCache.txt /therock/build/core/rocminfo/build/cmake_install.cmake /therock/build/compile_commands_fragment_rocminfo.json
256.2 cd /therock/build/core/rocminfo/build && /usr/bin/python3 /therock/src/build_tools/teatime.py --log-timestamps --label rocminfo\ configure --interactive /therock/build/logs/rocminfo_configure.log -- /usr/bin/cmake -GNinja -B/therock/build/core/rocminfo/build -S/therock/src/core/rocminfo -DCMAKE_INSTALL_PREFIX=/therock/build/core/rocminfo/stage -DCMAKE_TOOLCHAIN_FILE=/therock/build/core/rocminfo/_toolchain.cmake -DCMAKE_PROJECT_TOP_LEVEL_INCLUDES=/therock/build/core/rocminfo/_init.cmake && /usr/bin/cmake -E touch /therock/build/core/rocminfo/build/compile_commands.json && /usr/bin/cmake -E touch /therock/build/core/rocminfo/stamp/configure.stamp && /usr/bin/cmake -E copy /therock/build/core/rocminfo/build/compile_commands.json /therock/build/compile_commands_fragment_rocminfo.json
256.2 [rocminfo configure] -- Sub-project PKG_CONFIG_PATH: /therock/build/third-party/sysdeps/linux/bzip2/build/stage/lib/rocm_sysdeps/lib/pkgconfig:/therock/build/third-party/sysdeps/linux/zlib/build/stage/lib/rocm_sysdeps/lib/pkgconfig:/therock/build/third-party/sysdeps/linux/zstd/build/stage/lib/rocm_sysdeps/lib/pkgconfig:/therock/build/third-party/sysdeps/linux/elfutils/build/stage/lib/rocm_sysdeps/lib/pkgconfig:/therock/build/third-party/sysdeps/linux/libdrm/build/stage/lib/rocm_sysdeps/lib/pkgconfig:/therock/build/third-party/sysdeps/linux/numactl/build/stage/lib/rocm_sysdeps/lib/pkgconfig:
256.2 [rocminfo configure] -- Resolving packages from super-project: rocprofiler-register;BZip2;ZLIB;zstd;LibElf;libdw;NUMA;hsakmt;hsa-runtime64
256.2 [rocminfo configure] -- TheRock external project configuration injection (rocminfo)
256.2 [rocminfo configure] -- The C compiler identification is GNU 13.3.0
256.2 [rocminfo configure] -- The CXX compiler identification is GNU 13.3.0
256.2 [rocminfo configure] -- Detecting C compiler ABI info
256.2 [rocminfo configure] -- Detecting C compiler ABI info - done
256.2 [rocminfo configure] -- Check for working C compiler: /usr/bin/cc - skipped
256.2 [rocminfo configure] -- Detecting C compile features
256.2 [rocminfo configure] -- Detecting C compile features - done
256.2 [rocminfo configure] -- Detecting CXX compiler ABI info
256.2 [rocminfo configure] -- Detecting CXX compiler ABI info - done
256.2 [rocminfo configure] -- Check for working CXX compiler: /usr/bin/c++ - skipped
256.2 [rocminfo configure] -- Detecting CXX compile features
256.2 [rocminfo configure] -- Detecting CXX compile features - done
256.2 [rocminfo configure] CMake Error at CMakeLists.txt:57 (include):
256.2 [rocminfo configure]   include could not find requested file:
256.2 [rocminfo configure]
256.2 [rocminfo configure]     utils
256.2 [rocminfo configure]
256.2 [rocminfo configure]
256.2 [rocminfo configure] -- Resolving super-project find_package(hsa-runtime64 1.0 REQUIRED BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/core/ROCR-Runtime/dist/lib/cmake/hsa-runtime64)
256.2 [rocminfo configure] CMake Error at CMakeLists.txt:89 (get_package_version_number):
256.2 [rocminfo configure]   Unknown CMake command "get_package_version_number".
256.2 [rocminfo configure]
256.2 [rocminfo configure]
256.2 [rocminfo configure] -- Configuring incomplete, errors occurred!
256.2 [rocminfo configure completed in 0 seconds]
259.1 [201/303] Building sub-project hipify (in background)
259.1 [hipify] [1/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_CUB_API_functions.cpp.o
259.1 [hipify] [2/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_CAFFE2_API_types.cpp.o
259.1 [hipify] [3/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_Complex_API_types.cpp.o
259.1 [hipify] [4/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_CAFFE2_API_functions.cpp.o
259.1 [hipify] [5/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_RTC_API_types.cpp.o
259.1 [hipify] [6/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_Complex_API_functions.cpp.o
259.1 [hipify] [7/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_Device_types.cpp.o
259.1 [hipify] [8/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_CUB_API_types.cpp.o
259.1 [hipify] [9/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_FFT_API_types.cpp.o
259.1 [hipify] [10/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP.cpp.o
259.1 [hipify] [11/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_RAND_API_types.cpp.o
259.1 [hipify] [12/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_FFT_API_functions.cpp.o
259.1 [hipify] [13/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_TENSOR_API_functions.cpp.o
259.1 [hipify] [14/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_RAND_API_functions.cpp.o
259.1 [hipify] [15/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_RTC_API_functions.cpp.o
259.1 [hipify] [16/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_SOLVER_API_types.cpp.o
259.1 [hipify] [17/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_TENSOR_API_types.cpp.o
259.1 [hipify] [18/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_SPARSE_API_types.cpp.o
259.1 [hipify] [19/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_DNN_API_functions.cpp.o
259.1 [hipify] [20/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_Runtime_API_functions.cpp.o
259.1 [hipify] [21/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_SOLVER_API_functions.cpp.o
259.1 [hipify] [22/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_Driver_API_functions.cpp.o
259.1 [hipify] [23/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_Device_functions.cpp.o
259.1 [hipify] [24/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_Runtime_API_types.cpp.o
259.1 [hipify] [25/39] Building CXX object CMakeFiles/hipify-clang.dir/src/ArgParse.cpp.o
259.1 [hipify] [26/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_BLAS_API_types.cpp.o
259.1 [hipify] [27/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_DNN_API_types.cpp.o
259.1 [hipify] [28/39] Building CXX object CMakeFiles/hipify-clang.dir/src/Statistics.cpp.o
259.1 [hipify] [29/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_BLAS_API_functions.cpp.o
259.1 [hipify] [30/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_Driver_API_types.cpp.o
259.1 [hipify] [31/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_SPARSE_API_functions.cpp.o
259.1 [hipify] [32/39] Building CXX object CMakeFiles/hipify-clang.dir/src/StringUtils.cpp.o
259.1 [hipify] [33/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_Python.cpp.o
259.1 [hipify] [34/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_Doc.cpp.o
259.1 [hipify] [35/39] Building CXX object CMakeFiles/hipify-clang.dir/src/LLVMCompat.cpp.o
259.1 [hipify] [36/39] Building CXX object CMakeFiles/hipify-clang.dir/src/CUDA2HIP_Perl.cpp.o
259.1 [hipify] [37/39] Building CXX object CMakeFiles/hipify-clang.dir/src/main.cpp.o
259.1 [hipify] [38/39] Building CXX object CMakeFiles/hipify-clang.dir/src/HipifyAction.cpp.o
259.1 [hipify] [39/39] Linking CXX executable hipify-clang
259.1 [hipify completed in 11 seconds]
260.6 [202/303] Configure sub-project hipcc (in background)
260.6 [hipcc configure] -- Sub-project PKG_CONFIG_PATH: /therock/build/third-party/sysdeps/linux/zlib/build/stage/lib/rocm_sysdeps/lib/pkgconfig:/therock/build/third-party/sysdeps/linux/zstd/build/stage/lib/rocm_sysdeps/lib/pkgconfig:
260.6 [hipcc configure] -- Resolving packages from super-project: ZLIB;zstd;AMDDeviceLibs;Clang;LLD;LLVM;ROCmCMakeBuildTools;ROCM
260.6 [hipcc configure] -- TheRock external project configuration injection (hipcc)
260.6 [hipcc configure] -- The C compiler identification is GNU 13.3.0
260.6 [hipcc configure] -- The CXX compiler identification is GNU 13.3.0
260.6 [hipcc configure] -- Detecting C compiler ABI info
260.6 [hipcc configure] -- Detecting C compiler ABI info - done
260.6 [hipcc configure] -- Check for working C compiler: /usr/bin/cc - skipped
260.6 [hipcc configure] -- Detecting C compile features
260.6 [hipcc configure] -- Detecting C compile features - done
260.6 [hipcc configure] -- Detecting CXX compiler ABI info
260.6 [hipcc configure] -- Detecting CXX compiler ABI info - done
260.6 [hipcc configure] -- Check for working CXX compiler: /usr/bin/c++ - skipped
260.6 [hipcc configure] -- Detecting CXX compile features
260.6 [hipcc configure] -- Detecting CXX compile features - done
260.6 [hipcc configure] -- Resolving super-project find_package(ROCM QUIET BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/base/rocm-cmake/dist/share/rocm/cmake)
260.6 [hipcc configure] CMake Deprecation Warning at /therock/build/base/rocm-cmake/dist/share/rocm/cmake/ROCMConfig.cmake:12 (message):
260.6 [hipcc configure]   Use of find_package(ROCM) is deprecated as of ROCm 6.4.  Please use
260.6 [hipcc configure]   find_package(ROCmCMakeBuildTools)
260.6 [hipcc configure] Call Stack (most recent call first):
260.6 [hipcc configure]   /therock/src/cmake/therock_subproject_dep_provider.cmake:51 (find_package)
260.6 [hipcc configure]   CMakeLists.txt:8 (find_package)
260.6 [hipcc configure]
260.6 [hipcc configure]
260.6 [hipcc configure] -- Resolving super-project find_package(ROCmCMakeBuildTools BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/base/rocm-cmake/dist/share/rocmcmakebuildtools/cmake)
260.6 [hipcc configure] -- Configuring done (13.5s)
260.6 [hipcc configure] -- Generating done (0.0s)
260.6 [hipcc configure] -- Build files have been written to: /therock/build/compiler/hipcc/build
260.6 [hipcc configure completed in 13 seconds]
260.8 [203/303] Configure sub-project amd-comgr (in background)
260.8 [amd-comgr configure] Comgr Version: 2.9.0
260.8 [amd-comgr configure] -- Sub-project PKG_CONFIG_PATH: /therock/build/third-party/sysdeps/linux/zlib/build/stage/lib/rocm_sysdeps/lib/pkgconfig:/therock/build/third-party/sysdeps/linux/zstd/build/stage/lib/rocm_sysdeps/lib/pkgconfig:
260.8 [amd-comgr configure] -- Resolving packages from super-project: ZLIB;zstd;AMDDeviceLibs;Clang;LLD;LLVM;ROCmCMakeBuildTools;ROCM
260.8 [amd-comgr configure] -- TheRock external project configuration injection (amd_comgr)
260.8 [amd-comgr configure] -- The C compiler identification is GNU 13.3.0
260.8 [amd-comgr configure] -- The CXX compiler identification is GNU 13.3.0
260.8 [amd-comgr configure] -- Detecting C compiler ABI info
260.8 [amd-comgr configure] -- Detecting C compiler ABI info - done
260.8 [amd-comgr configure] -- Check for working C compiler: /usr/bin/cc - skipped
260.8 [amd-comgr configure] -- Detecting C compile features
260.8 [amd-comgr configure] -- Detecting C compile features - done
260.8 [amd-comgr configure] -- Detecting CXX compiler ABI info
260.8 [amd-comgr configure] -- Detecting CXX compiler ABI info - done
260.8 [amd-comgr configure] -- Check for working CXX compiler: /usr/bin/c++ - skipped
260.8 [amd-comgr configure] -- Detecting CXX compile features
260.8 [amd-comgr configure] -- Detecting CXX compile features - done
260.8 [amd-comgr configure] -- Resolving super-project find_package(ROCM QUIET BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/base/rocm-cmake/dist/share/rocm/cmake)
260.8 [amd-comgr configure] CMake Deprecation Warning at /therock/build/base/rocm-cmake/dist/share/rocm/cmake/ROCMConfig.cmake:12 (message):
260.8 [amd-comgr configure]   Use of find_package(ROCM) is deprecated as of ROCm 6.4.  Please use
260.8 [amd-comgr configure]   find_package(ROCmCMakeBuildTools)
260.8 [amd-comgr configure] Call Stack (most recent call first):
260.8 [amd-comgr configure]   /therock/src/cmake/therock_subproject_dep_provider.cmake:51 (find_package)
260.8 [amd-comgr configure]   CMakeLists.txt:53 (find_package)
260.8 [amd-comgr configure]
260.8 [amd-comgr configure]
260.8 [amd-comgr configure] -- Resolving super-project find_package(ROCmCMakeBuildTools BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/base/rocm-cmake/dist/share/rocmcmakebuildtools/cmake)
260.8 [amd-comgr configure] -- Resolving super-project find_package(AMDDeviceLibs REQUIRED BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/compiler/amd-llvm/dist/lib/llvm/lib/cmake/AMDDeviceLibs)
260.8 [amd-comgr configure] -- Resolving super-project find_package(Clang REQUIRED BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/compiler/amd-llvm/dist/lib/llvm/lib/cmake/clang)
260.8 [amd-comgr configure] -- Resolving super-project find_package(LLVM 19.0.0 EXACT REQUIRED BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/compiler/amd-llvm/dist/lib/llvm/lib/cmake/llvm)
260.8 [amd-comgr configure] -- Resolving super-project find_package(zstd BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/third-party/sysdeps/linux/zstd/build/dist/lib/rocm_sysdeps/lib/cmake/zstd)
260.8 [amd-comgr configure] -- Resolving super-project find_package(LLD REQUIRED BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/compiler/amd-llvm/dist/lib/llvm/lib/cmake/lld)
260.8 [amd-comgr configure] -- Resolving super-project find_package(LLVM 19.0.0 EXACT REQUIRED BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/compiler/amd-llvm/dist/lib/llvm/lib/cmake/llvm)
260.8 [amd-comgr configure] -- Resolving super-project find_package(zstd BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/third-party/sysdeps/linux/zstd/build/dist/lib/rocm_sysdeps/lib/cmake/zstd)
260.8 [amd-comgr configure]
260.8 [amd-comgr configure] ------------LLVM_DIR: /therock/build/compiler/amd-llvm/dist/lib/llvm/lib/cmake/llvm
260.8 [amd-comgr configure] ---LLVM_INCLUDE_DIRS: /therock/build/compiler/amd-llvm/dist/lib/llvm/include
260.8 [amd-comgr configure] ---LLVM_LIBRARY_DIRS: /therock/build/compiler/amd-llvm/dist/lib/llvm/lib
260.8 [amd-comgr configure] -----------Clang_DIR: /therock/build/compiler/amd-llvm/dist/lib/llvm/lib/cmake/clang
260.8 [amd-comgr configure] --CLANG_INCLUDE_DIRS: /therock/build/compiler/amd-llvm/dist/lib/llvm/include
260.8 [amd-comgr configure] ----LLD_INCLUDE_DIRS: /therock/build/compiler/amd-llvm/dist/lib/llvm/include
260.8 [amd-comgr configure] ---AMDDeviceLibs_DIR: /therock/build/compiler/amd-llvm/dist/lib/llvm/lib/cmake/AMDDeviceLibs
260.8 [amd-comgr configure] ------------ROCM_DIR: /therock/build/base/rocm-cmake/dist/share/rocm/cmake
260.8 [amd-comgr configure] ----COMGR_GIT_COMMIT: f26e3467aefc
260.8 [amd-comgr configure] ----COMGR_GIT_BRANCH: HEAD
260.8 [amd-comgr configure]
260.8 [amd-comgr configure] -- Comgr SPIRV Disabled
260.8 [amd-comgr configure] -- Resolving super-project find_package(Clang REQUIRED BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/compiler/amd-llvm/dist/lib/llvm/lib/cmake/clang)
260.8 [amd-comgr configure] -- Resolving super-project find_package(LLVM 19.0.0 EXACT REQUIRED BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/compiler/amd-llvm/dist/lib/llvm/lib/cmake/llvm)
260.8 [amd-comgr configure] -- Resolving super-project find_package(zstd BYPASS_PROVIDER NO_DEFAULT_PATH PATHS /therock/build/third-party/sysdeps/linux/zstd/build/dist/lib/rocm_sysdeps/lib/cmake/zstd)
260.8 [amd-comgr configure] -- Resolving system find_package(Python3) (not found in super-project ZLIB;zstd;AMDDeviceLibs;Clang;LLD;LLVM;ROCmCMakeBuildTools;ROCM)
260.8 [amd-comgr configure] -- Found Python3: /usr/bin/python3 (found version "3.12.3") found components: Interpreter
260.8 [amd-comgr configure] -- Resolving system find_package(Threads) (not found in super-project ZLIB;zstd;AMDDeviceLibs;Clang;LLD;LLVM;ROCmCMakeBuildTools;ROCM)
260.8 [amd-comgr configure] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
260.8 [amd-comgr configure] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
260.8 [amd-comgr configure] -- Found Threads: TRUE
260.8 [amd-comgr configure] -- Configuring done (13.7s)
260.8 [amd-comgr configure] -- Generating done (0.0s)
260.8 [amd-comgr configure] -- Build files have been written to: /therock/build/compiler/amd-comgr/build
260.8 [amd-comgr configure completed in 14 seconds]
260.8 ninja: build stopped: subcommand failed.
------
pytorch_dev_ubuntu_24.04.Dockerfile:24
--------------------
  22 |     # TODO: The ROCM components still output some things to the source dir. Remove
  23 |     # "rw" when fixed. See https://github.com/ROCm/TheRock/issues/159
  24 | >>> RUN --mount=type=bind,target=/therock/src,rw bash /build_rocm.sh "$AMDGPU_TARGETS"
  25 |
  26 |
--------------------
ERROR: failed to solve: process "/bin/sh -c bash /build_rocm.sh \"$AMDGPU_TARGETS\"" did not complete successfully: exit code: 1

hi @marbre Thanks for your efforts on it. I have a quick try of your branch and patch, and met a new issue of libdrm.

I check the log, and it showed that it failed to download libdrm package from https://gitlab.freedesktop.org/mesa/drm/-/archive/libdrm-2.4.124/drm-libdrm-2.4.124.tar.bz2 . I checked this link, and really didn't find the libdrm package. may we use the new link of https://dri.freedesktop.org/libdrm/libdrm-2.4.6.tar.bz2 instead of the above old link when downloading libdrm package? 

Thanks

4.528 CMake Error at therock-libdrm-sources_fetch/src/therock-libdrm-sources-stamp/therock-libdrm-sources-download-Release.cmake:37 (message):
4.528   Command failed: 1
4.528
4.528    '/usr/bin/cmake' '-Dmake=' '-Dconfig=' '-P' '/therock/build/third-party/sysdeps/linux/libdrm/therock-libdrm-sources_fetch/src/therock-libdrm-sources-stamp/therock-libdrm-sources-download-Release-impl.cmake'
4.528
4.528   See also
4.528
4.528     /therock/build/third-party/sysdeps/linux/libdrm/therock-libdrm-sources_fetch/src/therock-libdrm-sources-stamp/therock-libdrm-sources-download.log
4.528
4.528
4.528 -- Log output is:
4.528 -- Downloading...
4.528    dst='/therock/build/third-party/sysdeps/linux/libdrm/therock-libdrm-sources_fetch/src/drm-libdrm-2.4.124.tar.bz2'
4.528    timeout='none'
4.528    inactivity timeout='none'
4.528 -- Using src='https://gitlab.freedesktop.org/mesa/drm/-/archive/libdrm-2.4.124/drm-libdrm-2.4.124.tar.bz2'
4.528 CMake Error at therock-libdrm-sources_fetch/src/therock-libdrm-sources-stamp/download-therock-libdrm-sources.cmake:170 (message):
4.528   Each download failed!
4.528
4.528     error: downloading 'https://gitlab.freedesktop.org/mesa/drm/-/archive/libdrm-2.4.124/drm-libdrm-2.4.124.tar.bz2' failed
4.528           status_code: 22
4.528           status_string: "HTTP response code said error"
4.528           log:
4.528           --- LOG BEGIN ---
4.528           Host gitlab.freedesktop.org:443 was resolved.

I am seeing this issue as well, the error signature is as below:
vikagupt@AUS-Z16G2-VGUPT:/mnt/c/Users/vikagupt/Documents/GitHub/TheRock$ !553
cmake --build build
[0/2] Re-checking globbed directories...
[0/39] Building sub-project therock-aux-overlay
[therock-aux-overlay] [1/1] cd /mnt/c/Users/vikagupt/Documents/GitHub/TheRock/build/base/aux-overlay/build && /usr/local/lib/python3.10/dist-packages/cmake/data/bin/cmake -E create_symlink lib/llvm /mnt/c/Users/vikagupt/Documents/GitHub/TheRock/build/base/aux-overlay/build/llvm
[therock-aux-overlay completed in 0 seconds]
[6/39] Performing download step (download, verify and extract) for 'therock-libdrm-sources'
FAILED: third-party/sysdeps/linux/libdrm/therock-libdrm-sources_fetch/src/therock-libdrm-sources-stamp/therock-libdrm-sources-download /mnt/c/Users/vikagupt/Documents/GitHub/TheRock/build/third-party/sysdeps/linux/libdrm/therock-libdrm-sources_fetch/src/therock-libdrm-sources-stamp/therock-libdrm-sources-download
cd /mnt/c/Users/vikagupt/Documents/GitHub/TheRock/build/third-party/sysdeps/linux/libdrm && /usr/local/lib/python3.10/dist-packages/cmake/data/bin/cmake -P /mnt/c/Users/vikagupt/Documents/GitHub/TheRock/build/third-party/sysdeps/linux/libdrm/therock-libdrm-sources_fetch/src/therock-libdrm-sources-stamp/therock-libdrm-sources-download-Release.cmake && /usr/local/lib/python3.10/dist-packages/cmake/data/bin/cmake -E touch /mnt/c/Users/vikagupt/Documents/GitHub/TheRock/build/third-party/sysdeps/linux/libdrm/therock-libdrm-sources_fetch/src/therock-libdrm-sources-stamp/therock-libdrm-sources-download
CMake Error at therock-libdrm-sources_fetch/src/therock-libdrm-sources-stamp/therock-libdrm-sources-download-Release.cmake:37 (message):
  Command failed: 1

   '/usr/local/lib/python3.10/dist-packages/cmake/data/bin/cmake' '-Dmake=' '-Dconfig=' '-P' '/mnt/c/Users/vikagupt/Documents/GitHub/TheRock/build/third-party/sysdeps/linux/libdrm/therock-libdrm-sources_fetch/src/therock-libdrm-sources-stamp/therock-libdrm-sources-download-Release-impl.cmake'

  See also

    /mnt/c/Users/vikagupt/Documents/GitHub/TheRock/build/third-party/sysdeps/linux/libdrm/therock-libdrm-sources_fetch/src/therock-libdrm-sources-stamp/therock-libdrm-sources-download.log


-- Log output is:
-- verifying file...
       file='/mnt/c/Users/vikagupt/Documents/GitHub/TheRock/build/third-party/sysdeps/linux/libdrm/therock-libdrm-sources_fetch/src/drm-libdrm-2.4.124.tar.bz2'
-- SHA256 hash of
    /mnt/c/Users/vikagupt/Documents/GitHub/TheRock/build/third-party/sysdeps/linux/libdrm/therock-libdrm-sources_fetch/src/drm-libdrm-2.4.124.tar.bz2
  does not match expected value
    expected: '18e66044e0542040614a7904b6a2c0e5249a81e705fe9ba5a1cc2e5df11416e6'
      actual: 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
-- File already exists but hash mismatch. Removing...
-- Downloading...
   dst='/mnt/c/Users/vikagupt/Documents/GitHub/TheRock/build/third-party/sysdeps/linux/libdrm/therock-libdrm-sources_fetch/src/drm-libdrm-2.4.124.tar.bz2'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://gitlab.freedesktop.org/mesa/drm/-/archive/libdrm-2.4.124/drm-libdrm-2.4.124.tar.bz2'
CMake Error at therock-libdrm-sources_fetch/src/therock-libdrm-sources-stamp/download-therock-libdrm-sources.cmake:163 (message):
  Each download failed!

    error: downloading 'https://gitlab.freedesktop.org/mesa/drm/-/archive/libdrm-2.4.124/drm-libdrm-2.4.124.tar.bz2' failed
          status_code: 22
          status_string: "HTTP response code said error"
          log:
          --- LOG BEGIN ---
          timeout on name lookup is not supported

  Host gitlab.freedesktop.org:443 was resolved.

  IPv6: 2604:1380:45f2:6901::2

  IPv4: 147.28.146.59

    Trying [2604:1380:45f2:6901::2]:443...

  Immediate connect fail for 2604:1380:45f2:6901::2: Unknown error 101

--- a/patches/amd-mainline/rocminfo/0001-Use-idomatic-approach-to-extending-CMAKE_MODULE_PATH.patch
+++ b/patches/amd-mainline/rocminfo/0001-Use-idomatic-approach-to-extending-CMAKE_MODULE_PATH.patch
@@ -1,25 +1,29 @@
-From 8cb1187a0f684f2031d9cae7996efbb3fc4fa590 Mon Sep 17 00:00:00 2001
+From 03d7fd6c6b21150475700752b90d10c1255b11ff Mon Sep 17 00:00:00 2001
 From: Stella Laurenzo <stellaraccident@gmail.com>
 Date: Tue, 11 Feb 2025 22:10:22 -0800
 Subject: [PATCH] Use idomatic approach to extending CMAKE_MODULE_PATH.
 
-There are many reasons why there may already be a CMAKE_MODULE_PATH defined. The idiomatic way to extend it in a project is via `list(APPEND`.
+There are many reasons why there may already be a CMAKE_MODULE_PATH defined. The idiomatic way to extend it in a project is via `list(PREPEND`.
+
+Co-authored-by: Marius Brehler <marius.brehler@amd.com>
 ---
- CMakeLists.txt | 4 +---
- 1 file changed, 1 insertion(+), 3 deletions(-)
+ CMakeLists.txt | 6 ++----
+ 1 file changed, 2 insertions(+), 4 deletions(-)
 
 diff --git a/CMakeLists.txt b/CMakeLists.txt
-index df6755a..ecdcf77 100755
+index df6755a..f6d3a7e 100755
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -50,9 +50,7 @@ endif()
+@@ -49,10 +49,8 @@ endif()
+ # Generate static package, when BUILD_SHARED_LIBS is set to OFF.
  # Default to ON
  option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
- ## Set default module path if not already set
+-## Set default module path if not already set
 -if(NOT DEFINED CMAKE_MODULE_PATH)
 -    set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake_modules/")
 -endif()
-+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake_modules/")
++
++list(PREPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake_modules")
  ## Include common cmake modules
  include(utils)
  

